\documentclass[review]{elsarticle}

\usepackage{lineno,hyperref}
\modulolinenumbers[5]

\journal{Journal of \LaTeX\ Templates}

%%%%%%%%%%%%%%%%%%%%%%%
%% Elsevier bibliography styles
%%%%%%%%%%%%%%%%%%%%%%%
%% To change the style, put a % in front of the second line of the current style and
%% remove the % from the second line of the style you would like to use.
%%%%%%%%%%%%%%%%%%%%%%%

%% Numbered
%\bibliographystyle{model1-num-names}

%% Numbered without titles
%\bibliographystyle{model1a-num-names}

%% Harvard
%\bibliographystyle{model2-names.bst}\biboptions{authoryear}

%% Vancouver numbered
%\usepackage{numcompress}\bibliographystyle{model3-num-names}

%% Vancouver name/year
%\usepackage{numcompress}\bibliographystyle{model4-names}\biboptions{authoryear}

%% APA style
%\bibliographystyle{model5-names}\biboptions{authoryear}

%% AMA style
%\usepackage{numcompress}\bibliographystyle{model6-num-names}

%% `Elsevier LaTeX' style
\bibliographystyle{elsarticle-num}
%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\begin{frontmatter}

\title{Multi-SpaM: a Maximum-Likelihood approach to Phylogeny Reconstruction based on Multiple Spaced-Word Matches}
\tnotetext[mytitlenote]{Fully documented templates are available in the elsarticle package on \href{http://www.ctan.org/tex-archive/macros/latex/contrib/elsarticle}{CTAN}.}

%% Group authors per affiliation:
\author{Md. Nazmul Hasan }
\ead{0419052003}
\ead{nazmulcse25@gmail.com}
\fntext[myfootnote]{Since 1880.}

%% or include affiliations in footnotes:
\author{Arefin Rahman Niloy}
\ead{108052108}
\ead{arefinniloy@gmail.com}

\begin{abstract}
Word-based or ‘alignment-free’ methods for phylogeny reconstruction are much faster than traditional approaches, but they are generally less accurate. Most of these methods calculate
pairwise distances for a set of input sequences, for example from word
frequencies, from so-called spaced-word matches or from the average
length of common substrings. In this paper, we propose the first word-based approach
to tree reconstruction that is based on multiple sequence comparison
and Maximum Likelihood. Our algorithm first samples small, gap-free
alignments involving four taxa each. For each of these alignments, it 
 then calculates a quartet tree and, finally, the program Quartet MaxCut is used to infer a super tree topology for the full set of input taxa
from the calculated quartet trees. Experimental results show that
trees calculated with our approach are of high quality.
\end{abstract}

\begin{keyword}
\texttt{elsarticle.cls}\sep \LaTeX\sep Elsevier \sep template
\MSC[2010] 00-01\sep  99-00
\end{keyword}

\end{frontmatter}

\linenumbers

\section{The Elsevier article class}

\paragraph{} To gain a better understanding of the evolution of genes or species, reconstructing accurate phylogenetic trees is essential. This can be done using
standard methods which rely on sequence alignments, either of entire genomes
or of sets of orthologous genes or proteins. Character-based methods such as
Maximum Parsimony [14, 20] or Maximum Likelihood [15] infer trees based
on evolutionary substitution events that may have happened since the species
evolved from a common ancestor. These methods are generally considered to
be accurate, as long as the underlying alignments are of high quality, and as
long as suitable substitution models are used. However, for the task of multiple alignment no exact polynomial-time algorithm exists, and even heuristic
approaches can be time consuming [46]. Moreover, the most popular heuristic for multiple alignment, the progressive alignment [19], has been shown to
be relatively unstable: multiple alignments calculated with progressive approaches and trees inferred from these alignments depend on the underlying
guide trees and even on the order of the input sequences [9]. In addition to
these difficulties, exact algorithms for character-based phylogeny approaches
are themselves NP hard [11, 21].

\paragraph{Usage} Once the package is properly installed, you can use the document class \emph{elsarticle} to create a manuscript. Please make sure that your manuscript follows the guidelines in the Guide for Authors of the relevant journal. It is not necessary to typeset your manuscript in exactly the same way as an article, unless you are submitting to a camera-ready copy (CRC) journal.

\paragraph{Functionality} The Elsevier article class is based on the standard article class and supports almost all of the functionality of that class. In addition, it features commands and options to format the
\begin{itemize}
\item document style
\item baselineskip
\item front matter
\item keywords and MSC codes
\item theorems, definitions and proofs
\item lables of enumerations
\item citation style and labeling.
\end{itemize}

\section{Front matter}

The author names and affiliations could be formatted in two ways:
\begin{enumerate}[(1)]
\item Group the authors per affiliation.
\item Use footnotes to indicate the affiliations.
\end{enumerate}
See the front matter of this document for examples. You are recommended to conform your choice to the journal you are submitting to.

\section{Bibliography styles}

There are various bibliography styles available. You can select the style of your choice in the preamble of this document. These styles are Elsevier styles based on standard styles like Harvard and Vancouver. Please use Bib\TeX\ to generate your bibliography and include DOIs whenever available.

Here are two sample references: \cite{Feynman1963118,Dirac1953888}.

\section*{References}

\bibliography{mybibfile}

\end{document}